app/models/bishop.rb

```diff
@@ -7,4 +7,22 @@ def self.get_image(color)
       return "chess_piece_bishop_black.png"
     end
   end  
+
+
+ def valid_move?(x,y)
+    if diagonal_move?(x,y) && is_on_board?(x,y)
```

Comment:
It can be just one line `diagonal.. & is_on_board`.. will automatically return true/false :)